src/pages/Encounters/tabs/EncounterConsentsTab.tsx

```diff
@@ -0,0 +1,248 @@
+        </div>
+
+        <div className="flex justify-between items-start w-full gap-4 text-sm">
+          <div className="flex flex-col gap-1">
+            <span className="text-xs text-gray-500">{t("consent_date")}</span>
+            <p className="font-medium">
+              {formatDateTime(consent.date, "MMM D, YYYY")}
+            </p>
+          </div>
+          <div className="flex flex-col gap-1">
+            <span className="text-xs text-gray-500">{t("valid_period")}</span>
+            <p className="font-medium text-right">
+              <span>
+                {consent.period.start
+                  ? formatDateTime(consent.period.start, "MMM DD")
+                  : t("NA")}
+              </span>
+              {" - "}
+              <span>
+                {consent.period.end
+                  ? formatDateTime(consent.period.end, "MMM DD, YYYY")
+                  : t("NA")}
+              </span>
+            </p>
+          </div>
+        </div>
+      </CardContent>
+
+      <CardFooter className="p-0 border-t">
+        <Button
+          variant="ghost"
+          className="w-full justify-center items-center gap-2 rounded-t-none"
+          onClick={() =>
+            navigate(
+              `/facility/${facilityId}/patient/${encounter.patient.id}/encounter/${encounterId}/consents/${consentId}`,
+            )
+          }
+        >
+          <List className="size-4" />
+          {t("see_details")}
+        </Button>
+      </CardFooter>
+    </Card>
+  );
+}
+
+// Main tab component
+export const EncounterConsentsTab = ({ encounter }: EncounterTabProps) => {
+  const { t } = useTranslation();
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: CONSENTS_PER_PAGE,
+  });
+
+  const { data: existingConsents, isLoading } = useQuery({
+    queryKey: ["consents", encounter.patient.id, encounter.id, qParams],
+    queryFn: query(consentApi.list, {
+      pathParams: { patientId: encounter.patient.id },
+      queryParams: {
+        encounter: encounter.id,
+        limit: resultsPerPage,
+        offset: ((qParams.page || 1) - 1) * resultsPerPage,
+      },
+    }),
+  });
+
+  const filteredConsents = existingConsents?.results?.filter((consent) => {
+    if (!searchQuery) return true;
+
+    return consent.source_attachments.some((attachment) =>
+      attachment?.name?.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+  });
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+    updateQuery({ page: 1 });
+  };
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  return (
+    <div className="py-4">
+      <div className="flex flex-wrap items-center justify-between gap-4 mb-6">
+        <div className="relative flex-1 max-w-md">
+          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-muted-foreground size-4" />
```

Comment:
is this even a valid class as per our tailwind config? 🤔 

```suggestion
          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 size-4" />
```

Reply:
@rithviknishad 

> is this even a valid class as per our tailwind config? 🤔

I think no 

---

I just migrated that from https://github.com/ohcnetwork/care_fe/pull/12033/files#diff-7ea8fd0cf8f3235eda2a3a0da1d267386efc476e13807ffc1aa024d135bec4f5L117 (sheet to tab)

----

I see we are using it at multiple places 

![image](https://github.com/user-attachments/assets/351dff93-fe64-4e20-9952-1ee3afe13324)

---

is it coming from shadcn for shadcn components

Reply:
> is it coming from shadcn for shadcn components

it is coming from some AI written code that doesn't know what it's doing!

Reply:
@coderabbitai create an issue to clean other usages as well

AI creating mess, AI creating issues to clean up AI's mess :)